.github/workflows/build-executor-win.yaml

```diff
@@ -9,11 +9,27 @@ jobs:
   build:
     runs-on: windows-2022
     if: "!contains(github.event.head_commit.message, 'ci skip')"
+    env:
+      GO_REPOSITORY_USE_HOST_CACHE: 1
+      GOMODCACHE: "D:/go-mod-cache"
 
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Mount Bazel cache
+        uses: actions/cache@v4
+        with:
+          path: "D:/bazel/repo-cache/"
+          key: repo-cache
```

Comment:
Does this work? As far as I know GitHub would never invalidate the cache and just use the first one ever created.

Reply:
It does work to some extent.

Here https://github.com/buildbuddy-io/buildbuddy/actions/runs/13788951763/job/38563565843#step:3:18 is what we are currently running post-merge.

Here is a retry on the Windows workflow in this PR https://github.com/buildbuddy-io/buildbuddy/actions/runs/13788857479/job/38566023425#step:3:16

Im not too sure about invalidation tbh... but I think we rarely need to invalidate `--repository_cache=` right?

Reply:
I think that it works in the sense that the first run with a given key populates the cache and all future runs take that one snapshot. How about we invalidate this whenever the files change that also go into the Go module cache hash?

Reply:
@fmeum I think the suggestion is valid, but let's leave it for another PR so that we can change other workflows that are using the same key right now. Perhaps we can add a separate `restore-keys` similar to the Go cache.

@maggie-lou That's also a good suggestion but I would defer that to a later PR as well. Other workflows might benefit from the same change.

This PR is simply mirroring our existing setup in `.github/workflows/main.yaml` for the Windows workflows. So I would like to keep the scope small.